Developer data structures page

ivankocienski opened the pull request:
The documents, links and images covering how the database is structured

alexandria-gfsc commented on developers/data-structures.rst:
```diff
@@ -0,0 +1,84 @@
+
+Sites have a primary neighbourhood indicated by a field on one of the SiteNeighbourhood records. I don't think there is any logic stopping more than one site_neighbourhood from being flagged at a time.
+
+Neighbourhoods are recursively self-referential: this is how areas can contain areas like regions having many post-codes.
```
Hello :)

Not sure if it is worth mentioning that Neighbourhoods aren't infinitely recursive, but instead a bounded hierarchy. There's nothing actually stopping them from being an infinitely recursive tree, but it would break some assumptions about neighbourhood unit types and our dataset for them is now static and drawn from the official UK sources so it seems unlikely to change (except for eventual/possible porting to other countries' unit types :) )

Otherwise this seems good!

ivankocienski replied:
Hi,

Thank you for your feedback, I see your point. I think leaving that paragraph as is would accurately describe how the table works but a further paragraph describing the hierarchies would be useful.

alexandria-gfsc commented on developers/data-structures.rst:
```diff
@@ -0,0 +1,84 @@
+
+*(currently non-functional)*
+
+Tags allow Site admins (secretaries) to associate a Partner with a set of properties.
```
I think the usage of "secretaries" in the codebase is perhaps a vestigial term for site admin? I don't think it's used outside of some older components in a handful of comments, but not as an official term :) @kimadactyl maybe you can confirm?

ivankocienski replied:
I have started on a document that covers the mapping from the handbook glossary to how the code works. As far as I can tell users can either be "citizen" or "root". root being the place-cal admin and citizen being the Site admin

alexandria-gfsc replied:
Yeah! No that is fine, I'm just referring to the `(secretaries)` part. I'm not sure if that's a term we want to canonize since it's not used anywhere aside from a handful of comments -- that was why I tagged kim :)

alexandria-gfsc commented:
Okie! This is all great :) 

Just two small changes -- can you add a little bit about the Neighbourhoods hierarchy, and also change `secretaries` to `coordinators` or `event coordinators`. Reason for the latter part is because `secretaries` was originally part of the early PlaceCal vocab to refer to Site Admins / People using that part of the site, but it is really more of a specific term than what it refers to now, and it might just be a bit confusing for people reading it!

Thanks :)

ivankocienski commented:
Fixed the wording of secretaries -> coordinators and added a description of the Neighbourhood unit field.

Also apparently you are getting all the link fixes as well

alexandria-gfsc commented on commissioners/getting-started.rst:
```diff
@@ -15,7 +15,7 @@ Getting started with PlaceCal can be broken down into 7 steps.
 4. **Meet partners:** Arrange for your Secretaries to contact all
```
Oh lol, I missed it's presence here too. Maybe this is fine as it's less dev-docs-y.

alexandria-gfsc replied:
eh, itll be fine
